### PR TITLE
MNT-24096

### PR DESCRIPTION
**Pull request by aman-alfresco**
- disabled favorite feature below 23 version

### CODE REVIEW

File: actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt [hunk 1]
```diff
@@ -152,10 +153,14 @@ class ContextualActionsViewModel(
     private fun sharedActions(entry: Entry, entries: List<Entry>): List<Action> {
         val actions = mutableListOf<Action>()
         // Added Favorite Action
-        if (entries.any { !it.isFavorite }) {
-            actions.add(ActionAddFavorite(entry, entries))
-        } else {
-            actions.add(ActionRemoveFavorite(entry, entries))
+        val version = SearchRepository().getPrefsServerVersion()
+
+        if (version.toInt() >= 23) {
```
Comment (ankitgoyalgloballogic) [inline actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt:158]: Can we store version number 23 into a variable and then use it here?

File: app/src/main/java/com/alfresco/content/app/activity/LoginActivity.kt [hunk 2]
```diff
@@ -51,7 +52,7 @@ class LoginActivity : com.alfresco.auth.activity.LoginActivity() {
                 authConfig.jsonSerialize(),
                 endpoint,
                 person.displayName ?: "",
-                person.email,
+                person.email ?: "",
```
Comment (ankitgoyalgloballogic) [inline app/src/main/java/com/alfresco/content/app/activity/LoginActivity.kt:55]: Email is mandatory so no need to pass default value

File: data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt [hunk 3]
```diff
@@ -212,15 +216,68 @@ class SearchRepository {
...
+    fun getPrefsServerVersion(): String {
+        val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+        return sharedPrefs.getString(SERVER_VERSION, "") ?: ""
+    }
+
+    private fun saveServerVersion(serverVersion: String) {
+        val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+        val editor = sharedPrefs.edit()
+        editor.putString(SERVER_VERSION, serverVersion)
+        editor.apply()
```
Comment (ankitgoyalgloballogic) [inline data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt:253]: On logout, this saved data needs to be clear.

**Review by ankitgoyalgloballogic — CHANGES_REQUESTED** [review]

File: actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt (hunk 1 above)
Comment (aman-alfresco) [reply actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt:158]: Its using single time so global variable may create extra space in memory for this

File: data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt [hunk 4]
```diff
@@ -212,15 +216,68 @@ class SearchRepository {
...
-                session.account.id,
-                MAX_RECENT_FILES_AGE,
-                skipCount,
-                maxItems,
-            ),
-        )
+    suspend fun getRecents(skipCount: Int, maxItems: Int): ResponsePaging {
+        val version = getServerVersion()
+
+        if (version.toInt() >= 23) {
```
Comment (ankitgoyalgloballogic) [inline data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt:222]: Can we store version number 23 into a variable and then use it here?

**Review by ankitgoyalgloballogic — CHANGES_REQUESTED** [review]

File: actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt (hunk 1 above)
Comment (aman-alfresco) [reply actions/src/main/kotlin/com/alfresco/content/actions/ContextualActionsViewModel.kt]: done

File: app/src/main/java/com/alfresco/content/app/activity/LoginActivity.kt (hunk 2 above)
Comment (aman-alfresco) [reply app/src/main/java/com/alfresco/content/app/activity/LoginActivity.kt]: done

File: data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt (hunk 4 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt]: done

File: data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt (hunk 3 above)
Comment (aman-alfresco) [reply data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt:253]: done

**Review by ankitgoyalgloballogic — APPROVED** [review]

**Pull request merged**